arangod/Replication2/ReplicatedState/UpdateReplicatedState.cpp

```diff
@@ -74,10 +74,13 @@ auto algorithms::updateReplicatedState(
     return {TRI_ERROR_NO_ERROR};
   } else {
     auto status = state->getStatus();
-    TRI_ASSERT(status.has_value());
-    auto generation = status.value().getGeneration();
-    if (generation != expectedGeneration) {
-      state->flush(expectedGeneration);
+    if (status.has_value()) {
+      auto generation = status.value().getGeneration();
+      if (generation != expectedGeneration) {
+        state->flush(expectedGeneration);
+      }
+    } else if (!spec->participants.contains(serverId)) {
+      ctx.dropReplicatedState(id);
     }
```

Comment:
This is dead code, because [here](https://github.com/arangodb/arangodb/pull/16045/files?diff=split&w=1#diff-32db1afb5cbd60c1c1150796ca4256c98888a441993d6a9a21fc2c5dfd54b5fdL39) we already assert that the condition is true.